spark/src/main/scala/io/substrait/spark/logical/ToLogicalPlan.scala

```diff
@@ -157,11 +156,16 @@ class ToLogicalPlan(spark: SparkSession) extends DefaultRelVisitor[LogicalPlan]
   }
   override def visit(fetch: relation.Fetch): LogicalPlan = {
     val child = fetch.getInput.accept(this)
-    val limit = Literal(fetch.getCount.getAsLong.intValue(), IntegerType)
-    fetch.getOffset match {
-      case 1L => GlobalLimit(limitExpr = limit, child = child)
-      case -1L => LocalLimit(limitExpr = limit, child = child)
-      case _ => visitFallback(fetch)
+    if (fetch.getCount.isPresent) {
+      val limit = Literal(fetch.getCount.getAsLong.intValue(), IntegerType)
+      fetch.getOffset match {
+        case 1L => GlobalLimit(limitExpr = limit, child = child)
```

Comment:
fwiw this is just misusing the field, but I know it was there already so not a problem for this PR 😅

Reply:
If I understand this, you're using offset `1` or `-1` as a switching field? What's needed to not do this? At the very least, could we used something like `-2` to signal this to avoid using a valid `offset` value as a magic number?

Reply:
@Blizzara Do you know if there's anything outside of substrait-spark that's relying on this magic number, or is it something that we can change.

Reply:
I'm not aware of anything else, I'd be happy to see it changed! (I had already changed it in our internal fork of substrait-spark, but the code of this PR is nicer than mine 🎉 )